src/reporter.ts

```diff
@@ -127,4 +146,24 @@ export class Reporter {
       });
     });
   }
+
+  setLaunchStatus(status: string): void {
+    this.customLaunchStatus = status;
+  }
+
+  getCurrentSuiteId(): string {
+    return this.suiteIds.length ? this.suiteIds[this.suiteIds.length - 1] : undefined;
+  }
+
+  getCurrentTestItemId(): string {
+    return (
+      (this.testItems.length && this.testItems[this.testItems.length - 1].id) ||
+      this.getCurrentSuiteId()
+    );
+  }
+
+  setStatus({ status }: ObjUniversal): void {
+    const testItemId = this.getCurrentTestItemId();
+    this.testItemStatuses[testItemId] = status;
```

Comment:
Can we use `testData` or `testItems` entities to hold this info?
I think it would be better to keep all such information together (`testCaseId` and `attributes` will be introduced later) and not create a specific entity for each kind of data.